### PR TITLE
Core/Porting OMP parallel for loops to Parallel Utilities

### PR DESCRIPTION
**Pull request by ashishdarekar**
**Description**
Porting OMP parallel for loops to Parallel Utilities

- ParallelUtilities

**Changelog**
- Replaced the openmp for loops with block_for_each()

### CODE REVIEW

File: kratos/processes/apply_constant_scalarvalue_process.h [hunk 1]
```diff
@@ -379,16 +371,9 @@ class ApplyConstantScalarValueProcess : public Process
...
-//             ModelPart::NodesContainerType::iterator it_end = mr_model_part.GetMesh(mmesh_id).NodesEnd();
-
-             #pragma omp parallel for
-            for(int i = 0; i<nnodes; i++)
-            {
-                ModelPart::NodesContainerType::iterator it = it_begin + i;
-
-                it->FastGetSolutionStepValue(rVar) = value;
-            }
+            block_for_each(mr_model_part.GetMesh(mmesh_id).Nodes(), [&](Node<3>& rNode){
```
Comment (philbucher) [inline kratos/processes/apply_constant_scalarvalue_process.h]: I am pretty sure this also exists inthe VariableUtils

Comment (ashishdarekar) [reply kratos/processes/apply_constant_scalarvalue_process.h]: Yes, it is. Refactored :)

File: kratos/processes/apply_ray_casting_process.cpp [hunk 2]
```diff
@@ -70,15 +71,13 @@ namespace Kratos
 
 		ModelPart& ModelPart1 = mpFindIntersectedObjectsProcess->GetModelPart1();
 
-		#pragma omp parallel for
-		for(int k = 0 ; k < static_cast<int>(ModelPart1.NumberOfNodes()); ++k) {
-			auto it_node = ModelPart1.NodesBegin() + k;
-			double &r_node_distance = it_node->GetSolutionStepValue(DISTANCE);
-			const double ray_distance = this->DistancePositionInSpace(*it_node);
+		block_for_each(ModelPart1.Nodes(), [&](Node<3>& rNode){
+			double &r_node_distance = rNode.GetSolutionStepValue(DISTANCE);
```
Comment (philbucher) [inline kratos/processes/apply_ray_casting_process.cpp]:
```suggestion
			double &r_node_distance = rNode.FastGetSolutionStepValue(DISTANCE);
```

**Review by philbucher — APPROVED** [review]
looks good thx @ashishdarekar 

In the future you will split the header only processes into header & source for your HiWi work

**Pull request merged**